packages/xo-server/src/xapi/mixins/storage.mjs

```diff
@@ -74,17 +78,22 @@ export default {
     const vdis = this.getObject(sr).$VDIs
     const unhealthyVdis = { __proto__: null }
     const children = groupBy(vdis, 'sm_config.vhd-parent')
+    const vdisWithUnknownVhdParent = []
+
     const cache = { __proto__: null }
     forEach(vdis, vdi => {
       if (vdi.managed && !vdi.is_a_snapshot) {
         const { uuid } = vdi
-        const length = this._getUnhealthyVdiChainLength(uuid, children, cache)
+        const length = this._getUnhealthyVdiChainLength(uuid, children, cache, vdisWithUnknownVhdParent)
```

Comment:
It's a bit weird to get `vdisWithUnknownVhdParent` by passing it as a parameter. Maybe create an another function.